Bottom: 1b7f58c0568f7c9e2f46e74d6348c9f8a3e27a48
Top:    1cd885cc3a028001ab28944039250fb9cd655c2b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-06 17:15:46 +0000

Revert "Remove 511-character limit on configuration file lines"

This reverts commit d3830fdd06e18e05ef986210a3a9da2f49498195.

Tested-by: Kai-Martin Knaak <knaak@iqo.uni-hannover.de>


---

diff --git a/src/hid/gtk/gui-config.c b/src/hid/gtk/gui-config.c
index 0c48026..afbd575 100644
--- a/src/hid/gtk/gui-config.c
+++ b/src/hid/gtk/gui-config.c
@@ -474,14 +474,13 @@ static void
 config_file_read (void)
 {
   FILE *f;
-  char *buf = NULL;
-  size_t n = 0;
-  gchar *option, *arg;
+  gchar buf[512], *option, *arg;
 
   if ((f = config_file_open ("r")) == NULL)
     return;
 
-  while (getline (&buf, &n, f) != -1)
+  buf[0] = '\0';
+  while (fgets (buf, sizeof (buf), f))
     {
       if (parse_option_line (buf, &option, &arg) > 0)
 	set_config_attribute (option, arg);
@@ -489,7 +488,6 @@ config_file_read (void)
       g_free (arg);
     }
 
-  free (buf);
   fclose (f);
 }
